### DISCUSSION TITLE
Sum factorization

### DISCUSSION THREAD

**Discussion by hiemstar** [gridap/Gridap.jl#898] [General]
Gridap.jl looks like a great project. 

Is [sumfactorization](https://www.sam.math.ethz.ch/sam_reports/reports_final/reports1999/1999-15.pdf) implemented to evaluate integrals efficiently?

If not, it would be interesting to implement it, specifically combined with matrix-free methods and automatic differentiation, see e.g. [this paper](https://arxiv.org/abs/2204.01722)

**Comment by JordiManyer** [comment]
Hello @hiemstar, 
It is currently in the pipelines. I am working on matrix-free methods (including sum-factorization) as part of my PhD. I have a working prototype in a private repo, and hope to get a something more stable within the next few months that I can include in the public repos. 
So stay tuned, its coming 👍🏽

**Reply by hiemstar** [reply]
Nice! Maybe the following is helpful [feather](https://gitlab.com/feather-ecosystem/core/igaformation)

We have a Julia implementation with sum factorization for both matrix and matrix-free methods. Documentation is absent. If you want to know more let me know.
